{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set up notebook environment\n",
    "## NOTE: Use a QIIME2 kernel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import biom\n",
    "import warnings\n",
    "import pickle\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import qiime2 as q2\n",
    "from biom import Table\n",
    "from skbio import OrdinationResults\n",
    "from skbio.stats import subsample_counts\n",
    "from skbio.stats.distance import permanova, anosim, mantel\n",
    "from skbio.stats.distance import DistanceMatrix\n",
    "from qiime2.plugins.deicode.actions import rpca\n",
    "from qiime2.plugins.feature_table.actions import rarefy\n",
    "from qiime2.plugins.diversity.actions import beta_group_significance\n",
    "from qiime2.plugins.emperor.actions import biplot, plot\n",
    "from qiime2.plugins.diversity.actions import (beta,\n",
    "                                              beta_phylogenetic,\n",
    "                                              pcoa)\n",
    "from qiime2.plugins import demux, deblur, quality_filter, \\\n",
    "                           metadata, feature_table, alignment, \\\n",
    "                           phylogeny, diversity, emperor, feature_classifier, \\\n",
    "                           taxa, composition\n",
    "\n",
    "from assets.step_wise_anova import run_stepwise_anova\n",
    "from qiime2.plugins.fragment_insertion.actions import filter_features\n",
    "warnings.filterwarnings(\"ignore\", category=DeprecationWarning)\n",
    "\n",
    "# helper functions\n",
    "from assets.util_updated import (mantel_matched, simulate_depth,\n",
    "                        all_dists, all_dists_no_tree, nested_permanova)\n",
    "\n",
    "# plotting\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "plt.style.use('ggplot')\n",
    "%matplotlib inline\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subset metadata to make paired files between extraction kits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in sample metadata\n",
    "md = pd.read_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/12201_metadata.txt',\n",
    "                sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset sample metadata to make files for round 1 and round 2\n",
    "md_round1and2 = md[md['round'] != 3]\n",
    "md_round1 = md_round1and2[md_round1and2['round'] == 1]\n",
    "md_round2 = md_round1and2[md_round1and2['round'] == 2]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset round-specific metadata files to make files for each kit\n",
    "md_round1_powersoil = md_round1[md_round1['extraction_kit'] == 'PowerSoil']\n",
    "md_round1_powersoil_pro = md_round1[md_round1['extraction_kit'] == 'PowerSoil Pro']\n",
    "md_round1_norgen = md_round1[md_round1['extraction_kit'] == 'Norgen']\n",
    "md_round2_powersoil = md_round2[md_round2['extraction_kit'] == 'PowerSoil']\n",
    "md_round2_magmax = md_round2[md_round2['extraction_kit'] == 'MagMAX Microbiome']\n",
    "md_round2_nucleomag = md_round2[md_round2['extraction_kit'] == 'NucleoMag Food']\n",
    "md_round2_zymo = md_round2[md_round2['extraction_kit'] == 'Zymo MagBead']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge kit-specific files to make paired files for comparison\n",
    "md_round1_ps_vs_pro = pd.concat([md_round1_powersoil, md_round1_powersoil_pro])\n",
    "md_round1_ps_vs_norgen = pd.concat([md_round1_powersoil, md_round1_norgen])\n",
    "md_round2_ps_vs_magmax = pd.concat([md_round2_powersoil, md_round2_magmax])\n",
    "md_round2_ps_vs_nucleomag = pd.concat([md_round2_powersoil, md_round2_nucleomag])\n",
    "md_round2_ps_vs_zymo = pd.concat([md_round2_powersoil, md_round2_zymo])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export paired files\n",
    "md_round1_ps_vs_pro.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_pro.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round1_ps_vs_norgen.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_norgen.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_magmax.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_magmax.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_nucleomag.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_nucleomag.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_zymo.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_zymo.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mantel tests between pairs of kits"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 16S data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import data\n",
    "md_round1_ps_vs_pro_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_pro.txt')\n",
    "md_round1_ps_vs_norgen_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_norgen.txt')\n",
    "md_round2_ps_vs_magmax_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_magmax.txt')\n",
    "md_round2_ps_vs_nucleomag_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_nucleomag.txt')\n",
    "md_round2_ps_vs_zymo_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_zymo.txt')\n",
    "\n",
    "table_16S_hbm = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/16S/10_filtered_data/dna_bothPS_16S_deblur_biom_lod_noChl_noMit_sepp_gg_noNTCs_noMock_hbm.qza')\n",
    "table_16S_lbm = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/16S/10_filtered_data/dna_bothPS_16S_deblur_biom_lod_noChl_noMit_sepp_gg_noNTCs_noMock_lbm.qza')\n",
    "\n",
    "tree_16S = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/16S/09_fragment_insertion/dna_all_16S_deblur_seqs_noChl_noMit_tree_gg.qza')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.838001</td>\n",
       "      <td>0.789207</td>\n",
       "      <td>0.769943</td>\n",
       "      <td>0.84296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.00020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac      RPCA\n",
       "corr   0.838001            0.789207          0.769943   0.84296\n",
       "p      0.000200            0.000200          0.000200   0.00020\n",
       "n     45.000000           45.000000         45.000000  45.00000"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. PowerSoil Pro - High biomass samples\n",
    "## Filter table\n",
    "table_16S_hbm_biom = table_16S_hbm.view(Table)\n",
    "md_round1_ps_vs_pro_df_hbm = md_round1_ps_vs_pro_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_hbm_biom.ids()) & set(md_round1_ps_vs_pro_df_hbm.index))\n",
    "md_round1_ps_vs_pro_df_hbm = md_round1_ps_vs_pro_df_hbm.reindex(shared_)\n",
    "table_16S_hbm_biom_ps_vs_pro = table_16S_hbm_biom.filter(shared_)\n",
    "keep_ = table_16S_hbm_biom_ps_vs_pro.ids('observation')[table_16S_hbm_biom_ps_vs_pro.sum('observation') > 0]\n",
    "table_16S_hbm_biom_ps_vs_pro.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_hbm_ps_vs_pro = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_hbm_biom_ps_vs_pro)\n",
    "md_round1_ps_vs_pro_q2_hbm = q2.Metadata(md_round1_ps_vs_pro_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_hbm = 12690\n",
    "dists_res_16S_hbm = all_dists(table_16S_hbm_ps_vs_pro,\n",
    "                      rare_depth_16S_hbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_pro_q2_dist = md_round1_ps_vs_pro_q2_hbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_pro_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_pro_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_pro_q2_dist_sub = md_round1_ps_vs_pro_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_pro_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_hbm = pd.DataFrame(mantel_res_16S_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_hbm_ps_vs_pro.txt', sep='\\t')\n",
    "mantel_res_16S_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.907415</td>\n",
       "      <td>0.904647</td>\n",
       "      <td>0.90518</td>\n",
       "      <td>0.883156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.00020</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>28.000000</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>28.00000</td>\n",
       "      <td>28.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.907415            0.904647           0.90518   0.883156\n",
       "p      0.000200            0.000200           0.00020   0.000200\n",
       "n     28.000000           28.000000          28.00000  28.000000"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. PowerSoil Pro - Low biomass samples\n",
    "## Filter table\n",
    "table_16S_lbm_biom = table_16S_lbm.view(Table)\n",
    "md_round1_ps_vs_pro_df_lbm = md_round1_ps_vs_pro_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_lbm_biom.ids()) & set(md_round1_ps_vs_pro_df_lbm.index))\n",
    "md_round1_ps_vs_pro_df_lbm = md_round1_ps_vs_pro_df_lbm.reindex(shared_)\n",
    "table_16S_lbm_biom_ps_vs_pro = table_16S_lbm_biom.filter(shared_)\n",
    "keep_ = table_16S_lbm_biom_ps_vs_pro.ids('observation')[table_16S_lbm_biom_ps_vs_pro.sum('observation') > 0]\n",
    "table_16S_lbm_biom_ps_vs_pro.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_lbm_ps_vs_pro = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_lbm_biom_ps_vs_pro)\n",
    "md_round1_ps_vs_pro_q2_lbm = q2.Metadata(md_round1_ps_vs_pro_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_lbm = 3295\n",
    "dists_res_16S_lbm = all_dists(table_16S_lbm_ps_vs_pro,\n",
    "                      rare_depth_16S_lbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_pro_q2_dist = md_round1_ps_vs_pro_q2_lbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_pro_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_pro_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_pro_q2_dist_sub = md_round1_ps_vs_pro_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_pro_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_lbm = pd.DataFrame(mantel_res_16S_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_lbm_ps_vs_pro.txt', sep='\\t')\n",
    "mantel_res_16S_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.923284</td>\n",
       "      <td>0.873744</td>\n",
       "      <td>0.860568</td>\n",
       "      <td>0.860604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>42.000000</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>42.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.923284            0.873744          0.860568   0.860604\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     42.000000           42.000000         42.000000  42.000000"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Norgen - High biomass samples\n",
    "## Filter table\n",
    "table_16S_hbm_biom = table_16S_hbm.view(Table)\n",
    "md_round1_ps_vs_norgen_df_hbm = md_round1_ps_vs_norgen_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_hbm_biom.ids()) & set(md_round1_ps_vs_norgen_df_hbm.index))\n",
    "md_round1_ps_vs_norgen_df_hbm = md_round1_ps_vs_norgen_df_hbm.reindex(shared_)\n",
    "table_16S_hbm_biom_ps_vs_norgen = table_16S_hbm_biom.filter(shared_)\n",
    "keep_ = table_16S_hbm_biom_ps_vs_norgen.ids('observation')[table_16S_hbm_biom_ps_vs_norgen.sum('observation') > 0]\n",
    "table_16S_hbm_biom_ps_vs_norgen.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_hbm_ps_vs_norgen = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_hbm_biom_ps_vs_norgen)\n",
    "md_round1_ps_vs_norgen_q2_hbm = q2.Metadata(md_round1_ps_vs_norgen_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_hbm = 12690\n",
    "dists_res_16S_hbm = all_dists(table_16S_hbm_ps_vs_norgen,\n",
    "                      rare_depth_16S_hbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_norgen_q2_dist = md_round1_ps_vs_norgen_q2_hbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_norgen_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_norgen_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_norgen_q2_dist_sub = md_round1_ps_vs_norgen_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_norgen_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_hbm = pd.DataFrame(mantel_res_16S_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_hbm_ps_vs_norgen.txt', sep='\\t')\n",
    "mantel_res_16S_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.81055</td>\n",
       "      <td>0.327616</td>\n",
       "      <td>0.850769</td>\n",
       "      <td>0.070740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.00220</td>\n",
       "      <td>0.211158</td>\n",
       "      <td>0.007598</td>\n",
       "      <td>0.777045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>7.00000</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>7.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Jaccard  Unweighted UniFrac  Weighted UniFrac      RPCA\n",
       "corr  0.81055            0.327616          0.850769  0.070740\n",
       "p     0.00220            0.211158          0.007598  0.777045\n",
       "n     7.00000            7.000000          7.000000  7.000000"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Norgen - High biomass samples\n",
    "## Filter table\n",
    "table_16S_lbm_biom = table_16S_lbm.view(Table)\n",
    "md_round1_ps_vs_norgen_df_lbm = md_round1_ps_vs_norgen_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_lbm_biom.ids()) & set(md_round1_ps_vs_norgen_df_lbm.index))\n",
    "md_round1_ps_vs_norgen_df_lbm = md_round1_ps_vs_norgen_df_lbm.reindex(shared_)\n",
    "table_16S_lbm_biom_ps_vs_norgen = table_16S_lbm_biom.filter(shared_)\n",
    "keep_ = table_16S_lbm_biom_ps_vs_norgen.ids('observation')[table_16S_lbm_biom_ps_vs_norgen.sum('observation') > 0]\n",
    "table_16S_lbm_biom_ps_vs_norgen.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_lbm_ps_vs_norgen = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_lbm_biom_ps_vs_norgen)\n",
    "md_round1_ps_vs_norgen_q2_lbm = q2.Metadata(md_round1_ps_vs_norgen_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_lbm = 3295\n",
    "dists_res_16S_lbm = all_dists(table_16S_lbm_ps_vs_norgen,\n",
    "                      rare_depth_16S_lbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_norgen_q2_dist = md_round1_ps_vs_norgen_q2_lbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_norgen_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_norgen_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_norgen_q2_dist_sub = md_round1_ps_vs_norgen_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_norgen_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_lbm = pd.DataFrame(mantel_res_16S_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_lbm_ps_vs_norgen.txt', sep='\\t')\n",
    "mantel_res_16S_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.954889</td>\n",
       "      <td>0.895365</td>\n",
       "      <td>0.931581</td>\n",
       "      <td>0.904735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>40.000000</td>\n",
       "      <td>40.000000</td>\n",
       "      <td>40.000000</td>\n",
       "      <td>40.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.954889            0.895365          0.931581   0.904735\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     40.000000           40.000000         40.000000  40.000000"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. MagMAX Microbiome - High biomass samples\n",
    "## Filter table\n",
    "table_16S_hbm_biom = table_16S_hbm.view(Table)\n",
    "md_round2_ps_vs_magmax_df_hbm = md_round2_ps_vs_magmax_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_hbm_biom.ids()) & set(md_round2_ps_vs_magmax_df_hbm.index))\n",
    "md_round2_ps_vs_magmax_df_hbm = md_round2_ps_vs_magmax_df_hbm.reindex(shared_)\n",
    "table_16S_hbm_biom_ps_vs_magmax = table_16S_hbm_biom.filter(shared_)\n",
    "keep_ = table_16S_hbm_biom_ps_vs_magmax.ids('observation')[table_16S_hbm_biom_ps_vs_magmax.sum('observation') > 0]\n",
    "table_16S_hbm_biom_ps_vs_magmax.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_hbm_ps_vs_magmax = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_hbm_biom_ps_vs_magmax)\n",
    "md_round2_ps_vs_magmax_q2_hbm = q2.Metadata(md_round2_ps_vs_magmax_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_hbm = 12690\n",
    "dists_res_16S_hbm = all_dists(table_16S_hbm_ps_vs_magmax,\n",
    "                      rare_depth_16S_hbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_magmax_q2_dist = md_round2_ps_vs_magmax_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_magmax_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_magmax_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_magmax_q2_dist_sub = md_round2_ps_vs_magmax_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_magmax_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_hbm = pd.DataFrame(mantel_res_16S_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_hbm_ps_vs_magmax.txt', sep='\\t')\n",
    "mantel_res_16S_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.912166</td>\n",
       "      <td>0.922532</td>\n",
       "      <td>0.926807</td>\n",
       "      <td>0.831227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>48.000000</td>\n",
       "      <td>48.000000</td>\n",
       "      <td>48.000000</td>\n",
       "      <td>48.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.912166            0.922532          0.926807   0.831227\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     48.000000           48.000000         48.000000  48.000000"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. MagMAX Microbiome - Low biomass samples\n",
    "## Filter table\n",
    "table_16S_lbm_biom = table_16S_lbm.view(Table)\n",
    "md_round2_ps_vs_magmax_df_lbm = md_round2_ps_vs_magmax_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_lbm_biom.ids()) & set(md_round2_ps_vs_magmax_df_lbm.index))\n",
    "md_round2_ps_vs_magmax_df_lbm = md_round2_ps_vs_magmax_df_lbm.reindex(shared_)\n",
    "table_16S_lbm_biom_ps_vs_magmax = table_16S_lbm_biom.filter(shared_)\n",
    "keep_ = table_16S_lbm_biom_ps_vs_magmax.ids('observation')[table_16S_lbm_biom_ps_vs_magmax.sum('observation') > 0]\n",
    "table_16S_lbm_biom_ps_vs_magmax.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_lbm_ps_vs_magmax = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_lbm_biom_ps_vs_magmax)\n",
    "md_round2_ps_vs_magmax_q2_lbm = q2.Metadata(md_round2_ps_vs_magmax_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_lbm = 3295\n",
    "dists_res_16S_lbm = all_dists(table_16S_lbm_ps_vs_magmax,\n",
    "                      rare_depth_16S_lbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_magmax_q2_dist = md_round2_ps_vs_magmax_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_magmax_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_magmax_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_magmax_q2_dist_sub = md_round2_ps_vs_magmax_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_magmax_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_lbm = pd.DataFrame(mantel_res_16S_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_lbm_ps_vs_magmax.txt', sep='\\t')\n",
    "mantel_res_16S_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.956863</td>\n",
       "      <td>0.896437</td>\n",
       "      <td>0.939377</td>\n",
       "      <td>0.928647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.956863            0.896437          0.939377   0.928647\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     45.000000           45.000000         45.000000  45.000000"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. NucleoMag Food - High biomass samples\n",
    "## Filter table\n",
    "table_16S_hbm_biom = table_16S_hbm.view(Table)\n",
    "md_round2_ps_vs_nucleomag_df_hbm = md_round2_ps_vs_nucleomag_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_hbm_biom.ids()) & set(md_round2_ps_vs_nucleomag_df_hbm.index))\n",
    "md_round2_ps_vs_nucleomag_df_hbm = md_round2_ps_vs_nucleomag_df_hbm.reindex(shared_)\n",
    "table_16S_hbm_biom_ps_vs_nucleomag = table_16S_hbm_biom.filter(shared_)\n",
    "keep_ = table_16S_hbm_biom_ps_vs_nucleomag.ids('observation')[table_16S_hbm_biom_ps_vs_nucleomag.sum('observation') > 0]\n",
    "table_16S_hbm_biom_ps_vs_nucleomag.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_hbm_ps_vs_nucleomag = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_hbm_biom_ps_vs_nucleomag)\n",
    "md_round2_ps_vs_nucleomag_q2_hbm = q2.Metadata(md_round2_ps_vs_nucleomag_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_hbm = 12690\n",
    "dists_res_16S_hbm = all_dists(table_16S_hbm_ps_vs_nucleomag,\n",
    "                      rare_depth_16S_hbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_nucleomag_q2_dist = md_round2_ps_vs_nucleomag_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_nucleomag_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_nucleomag_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_nucleomag_q2_dist_sub = md_round2_ps_vs_nucleomag_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_nucleomag_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_hbm = pd.DataFrame(mantel_res_16S_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_hbm_ps_vs_nucleomag.txt', sep='\\t')\n",
    "mantel_res_16S_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.891625</td>\n",
       "      <td>0.897431</td>\n",
       "      <td>0.897413</td>\n",
       "      <td>0.818539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>44.000000</td>\n",
       "      <td>44.000000</td>\n",
       "      <td>44.000000</td>\n",
       "      <td>44.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.891625            0.897431          0.897413   0.818539\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     44.000000           44.000000         44.000000  44.000000"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. NucleoMag Food - Low biomass samples\n",
    "## Filter table\n",
    "table_16S_lbm_biom = table_16S_lbm.view(Table)\n",
    "md_round2_ps_vs_nucleomag_df_lbm = md_round2_ps_vs_nucleomag_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_lbm_biom.ids()) & set(md_round2_ps_vs_nucleomag_df_lbm.index))\n",
    "md_round2_ps_vs_nucleomag_df_lbm = md_round2_ps_vs_nucleomag_df_lbm.reindex(shared_)\n",
    "table_16S_lbm_biom_ps_vs_nucleomag = table_16S_lbm_biom.filter(shared_)\n",
    "keep_ = table_16S_lbm_biom_ps_vs_nucleomag.ids('observation')[table_16S_lbm_biom_ps_vs_nucleomag.sum('observation') > 0]\n",
    "table_16S_lbm_biom_ps_vs_nucleomag.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_lbm_ps_vs_nucleomag = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_lbm_biom_ps_vs_nucleomag)\n",
    "md_round2_ps_vs_nucleomag_q2_lbm = q2.Metadata(md_round2_ps_vs_nucleomag_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_lbm = 3295\n",
    "dists_res_16S_lbm = all_dists(table_16S_lbm_ps_vs_nucleomag,\n",
    "                      rare_depth_16S_lbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_nucleomag_q2_dist = md_round2_ps_vs_nucleomag_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_nucleomag_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_nucleomag_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_nucleomag_q2_dist_sub = md_round2_ps_vs_nucleomag_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_nucleomag_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_lbm = pd.DataFrame(mantel_res_16S_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_lbm_ps_vs_nucleomag.txt', sep='\\t')\n",
    "mantel_res_16S_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.964286</td>\n",
       "      <td>0.906934</td>\n",
       "      <td>0.901048</td>\n",
       "      <td>0.960285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.964286            0.906934          0.901048   0.960285\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     45.000000           45.000000         45.000000  45.000000"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Zymo MagBead - High biomass samples\n",
    "## Filter table\n",
    "table_16S_hbm_biom = table_16S_hbm.view(Table)\n",
    "md_round2_ps_vs_zymo_df_hbm = md_round2_ps_vs_zymo_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_hbm_biom.ids()) & set(md_round2_ps_vs_zymo_df_hbm.index))\n",
    "md_round2_ps_vs_zymo_df_hbm = md_round2_ps_vs_zymo_df_hbm.reindex(shared_)\n",
    "table_16S_hbm_biom_ps_vs_zymo = table_16S_hbm_biom.filter(shared_)\n",
    "keep_ = table_16S_hbm_biom_ps_vs_zymo.ids('observation')[table_16S_hbm_biom_ps_vs_zymo.sum('observation') > 0]\n",
    "table_16S_hbm_biom_ps_vs_zymo.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_hbm_ps_vs_zymo = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_hbm_biom_ps_vs_zymo)\n",
    "md_round2_ps_vs_zymo_q2_hbm = q2.Metadata(md_round2_ps_vs_zymo_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_hbm = 12690\n",
    "dists_res_16S_hbm = all_dists(table_16S_hbm_ps_vs_zymo,\n",
    "                      rare_depth_16S_hbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_zymo_q2_dist = md_round2_ps_vs_zymo_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_zymo_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_zymo_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_zymo_q2_dist_sub = md_round2_ps_vs_zymo_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_zymo_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_hbm = pd.DataFrame(mantel_res_16S_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_hbm_ps_vs_zymo.txt', sep='\\t')\n",
    "mantel_res_16S_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>Unweighted UniFrac</th>\n",
       "      <th>Weighted UniFrac</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.892554</td>\n",
       "      <td>0.888277</td>\n",
       "      <td>0.866754</td>\n",
       "      <td>0.810922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>18.000000</td>\n",
       "      <td>18.000000</td>\n",
       "      <td>18.000000</td>\n",
       "      <td>18.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard  Unweighted UniFrac  Weighted UniFrac       RPCA\n",
       "corr   0.892554            0.888277          0.866754   0.810922\n",
       "p      0.000200            0.000200          0.000200   0.000200\n",
       "n     18.000000           18.000000         18.000000  18.000000"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Zymo MagBead - High biomass samples\n",
    "## Filter table\n",
    "table_16S_lbm_biom = table_16S_lbm.view(Table)\n",
    "md_round2_ps_vs_zymo_df_lbm = md_round2_ps_vs_zymo_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_lbm_biom.ids()) & set(md_round2_ps_vs_zymo_df_lbm.index))\n",
    "md_round2_ps_vs_zymo_df_lbm = md_round2_ps_vs_zymo_df_lbm.reindex(shared_)\n",
    "table_16S_lbm_biom_ps_vs_zymo = table_16S_lbm_biom.filter(shared_)\n",
    "keep_ = table_16S_lbm_biom_ps_vs_zymo.ids('observation')[table_16S_lbm_biom_ps_vs_zymo.sum('observation') > 0]\n",
    "table_16S_lbm_biom_ps_vs_zymo.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_16S_lbm_ps_vs_zymo = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_lbm_biom_ps_vs_zymo)\n",
    "md_round2_ps_vs_zymo_q2_lbm = q2.Metadata(md_round2_ps_vs_zymo_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_16S_lbm = 3295\n",
    "dists_res_16S_lbm = all_dists(table_16S_lbm_ps_vs_zymo,\n",
    "                      rare_depth_16S_lbm, tree_16S)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_zymo_q2_dist = md_round2_ps_vs_zymo_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_zymo_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_zymo_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_16S_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_16S_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_zymo_q2_dist_sub = md_round2_ps_vs_zymo_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_16S_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_zymo_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_16S_lbm = pd.DataFrame(mantel_res_16S_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_16S_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_16S_lbm_ps_vs_zymo.txt', sep='\\t')\n",
    "mantel_res_16S_lbm\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Stepwise ANOVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sample_name\n",
       "12201.built.keyboard.A.1.2.zymoMA.L    12201.built.keyboard.A.1.2\n",
       "12201.urine.male.C.2.powersoil.L             12201.urine.male.C.2\n",
       "12201.skin.armpit.C.1.zymoMA.L              12201.skin.armpit.C.1\n",
       "12201.human.milk.Z.1.powersoil.L             12201.human.milk.Z.1\n",
       "12201.urine.female.D.1.powersoil.L         12201.urine.female.D.1\n",
       "                                                  ...            \n",
       "12201.urine.male.B.3.powersoil.L             12201.urine.male.B.3\n",
       "12201.skin.armpit.B.2.powersoil.L           12201.skin.armpit.B.2\n",
       "12201.urine.female.E.2.powersoil.L         12201.urine.female.E.2\n",
       "12201.built.keyboard.A.1.1.zymoMA.L    12201.built.keyboard.A.1.1\n",
       "12201.urine.female.F.3.zymoMA.L            12201.urine.female.F.3\n",
       "Name: unique_sample_id, Length: 96, dtype: object"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "md_round2_ps_vs_zymo_q2_dist[ids]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate ordinations (row=samples, cols=axes)\n",
    "pcoa_res = {}\n",
    "pcoa_res['Jaccard'] = pcoa(dists_res_16S_lbm['Jaccard'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['Unweighted UniFrac'] = pcoa(dists_res_16S_lbm['Unweighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['Weighted UniFrac'] = pcoa(dists_res_16S_lbm['Weighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['RPCA'] = dists_res_16S_lbm['RPCA'].biplot.view(OrdinationResults).samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "es_all = {}\n",
    "use_ = ['sample_type', 'sample_type_2','sample_type_3','biomass_sample','sample_technical_replicate', 'bead_beating']\n",
    "# clean up meta (only stuff to run)\n",
    "mf_ord = mf.to_dataframe().copy()\n",
    "# shit filter but works for now\n",
    "keep_ = [v_ for v_ in mf_ord.columns\n",
    "         if len(set(mf_ord[v_])) > 1 and\n",
    "         len(set(mf_ord[v_])) < mf_ord.shape[0]//2]\n",
    "mf_ord = mf_ord[keep_]\n",
    "# run stp-wise ANOVA for all ords\n",
    "for metric_, ord_ in  pcoa_res.items():\n",
    "    # get first three axes\n",
    "    ord_ = ord_[[0,1,2]]\n",
    "    ord_.columns = ['PC1','PC2','PC3']\n",
    "    # subset/match\n",
    "    mf_ord_ = mf_ord.copy()\n",
    "    shared_ids = list(set(ord_.index)\\\n",
    "                      & set(mf_ord_.index))\n",
    "    mf_ord_ = mf_ord_.loc[shared_ids,:]\n",
    "    ord_ = ord_.loc[shared_ids,:]\n",
    "    es_all[metric_] = run_stepwise_anova(ord_, mf_ord_, use_) #mf_ord_.columns)\n",
    "# concat all runs\n",
    "es_alldf = pd.concat(es_all).rename({'+ sample_type_2':'Sample Type'}, axis=0)\n",
    "es_alldf.to_csv('results/tables/effect-size_2min_20min.tsv', sep='\\t')\n",
    "es_alldf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There appears to be differences between bead_beating, biomass, and incubation with various metrics."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (qiime2-2021.11)",
   "language": "python",
   "name": "qiime2-2021.11"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
