{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set up notebook environment\n",
    "## NOTE: Use a QIIME2 kernel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import biom\n",
    "import warnings\n",
    "import pickle\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import qiime2 as q2\n",
    "from biom import Table\n",
    "from skbio import OrdinationResults\n",
    "from skbio.stats import subsample_counts\n",
    "from skbio.stats.distance import permanova, anosim, mantel\n",
    "from skbio.stats.distance import DistanceMatrix\n",
    "from qiime2.plugins.deicode.actions import rpca\n",
    "from qiime2.plugins.feature_table.actions import rarefy\n",
    "from qiime2.plugins.diversity.actions import beta_group_significance\n",
    "from qiime2.plugins.emperor.actions import biplot, plot\n",
    "from qiime2.plugins.diversity.actions import (beta,\n",
    "                                              beta_phylogenetic,\n",
    "                                              pcoa)\n",
    "from qiime2.plugins import demux, deblur, quality_filter, \\\n",
    "                           metadata, feature_table, alignment, \\\n",
    "                           phylogeny, diversity, emperor, feature_classifier, \\\n",
    "                           taxa, composition\n",
    "\n",
    "from assets.step_wise_anova import run_stepwise_anova\n",
    "from qiime2.plugins.fragment_insertion.actions import filter_features\n",
    "warnings.filterwarnings(\"ignore\", category=DeprecationWarning)\n",
    "\n",
    "# helper functions\n",
    "from assets.util_updated import (mantel_matched, simulate_depth,\n",
    "                        all_dists, all_dists_no_tree, nested_permanova)\n",
    "\n",
    "# plotting\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "plt.style.use('ggplot')\n",
    "%matplotlib inline\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subset metadata to make paired files between extraction kits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in sample metadata\n",
    "md = pd.read_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/12201_metadata.txt',\n",
    "                sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset sample metadata to make files for round 1 and round 2\n",
    "md_round1and2 = md[md['round'] != 3]\n",
    "md_round1 = md_round1and2[md_round1and2['round'] == 1]\n",
    "md_round2 = md_round1and2[md_round1and2['round'] == 2]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset round-specific metadata files to make files for each kit\n",
    "md_round1_powersoil = md_round1[md_round1['extraction_kit'] == 'PowerSoil']\n",
    "md_round1_powersoil_pro = md_round1[md_round1['extraction_kit'] == 'PowerSoil Pro']\n",
    "md_round1_norgen = md_round1[md_round1['extraction_kit'] == 'Norgen']\n",
    "md_round2_powersoil = md_round2[md_round2['extraction_kit'] == 'PowerSoil']\n",
    "md_round2_magmax = md_round2[md_round2['extraction_kit'] == 'MagMAX Microbiome']\n",
    "md_round2_nucleomag = md_round2[md_round2['extraction_kit'] == 'NucleoMag Food']\n",
    "md_round2_zymo = md_round2[md_round2['extraction_kit'] == 'Zymo MagBead']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge kit-specific files to make paired files for comparison\n",
    "md_round1_ps_vs_pro = pd.concat([md_round1_powersoil, md_round1_powersoil_pro])\n",
    "md_round1_ps_vs_norgen = pd.concat([md_round1_powersoil, md_round1_norgen])\n",
    "md_round2_ps_vs_magmax = pd.concat([md_round2_powersoil, md_round2_magmax])\n",
    "md_round2_ps_vs_nucleomag = pd.concat([md_round2_powersoil, md_round2_nucleomag])\n",
    "md_round2_ps_vs_zymo = pd.concat([md_round2_powersoil, md_round2_zymo])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export paired files\n",
    "md_round1_ps_vs_pro.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_pro.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round1_ps_vs_norgen.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_norgen.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_magmax.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_magmax.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_nucleomag.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_nucleomag.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n",
    "md_round2_ps_vs_zymo.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_zymo.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mantel tests between pairs of kits"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ITS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import data\n",
    "md_round1_ps_vs_pro_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_pro.txt')\n",
    "md_round1_ps_vs_norgen_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1_ps_vs_norgen.txt')\n",
    "md_round2_ps_vs_magmax_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_magmax.txt')\n",
    "md_round2_ps_vs_nucleomag_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_nucleomag.txt')\n",
    "md_round2_ps_vs_zymo_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round2_ps_vs_zymo.txt')\n",
    "\n",
    "table_its_hbm = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/ITS/08_filtered_data/dna_bothPS_ITS_deblur_biom_lod_noNTCs_noMock_hbm.qza')\n",
    "table_its_lbm = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/ITS/08_filtered_data/dna_bothPS_ITS_deblur_biom_lod_noNTCs_noMock_lbm.qza')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.802387</td>\n",
       "      <td>0.772924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>25.000000</td>\n",
       "      <td>25.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.802387   0.772924\n",
       "p      0.000200   0.000200\n",
       "n     25.000000  25.000000"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. PowerSoil Pro - High biomass samples\n",
    "## Filter table\n",
    "table_its_hbm_biom = table_its_hbm.view(Table)\n",
    "md_round1_ps_vs_pro_df_hbm = md_round1_ps_vs_pro_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_hbm_biom.ids()) & set(md_round1_ps_vs_pro_df_hbm.index))\n",
    "md_round1_ps_vs_pro_df_hbm = md_round1_ps_vs_pro_df_hbm.reindex(shared_)\n",
    "table_its_hbm_biom_ps_vs_pro = table_its_hbm_biom.filter(shared_)\n",
    "keep_ = table_its_hbm_biom_ps_vs_pro.ids('observation')[table_its_hbm_biom_ps_vs_pro.sum('observation') > 0]\n",
    "table_its_hbm_biom_ps_vs_pro.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_hbm_ps_vs_pro = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_hbm_biom_ps_vs_pro)\n",
    "md_round1_ps_vs_pro_q2_hbm = q2.Metadata(md_round1_ps_vs_pro_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_hbm = 1491\n",
    "dists_res_its_hbm = all_dists_no_tree(table_its_hbm_ps_vs_pro,\n",
    "                      rare_depth_its_hbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_pro_q2_dist = md_round1_ps_vs_pro_q2_hbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_pro_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_pro_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_pro_q2_dist_sub = md_round1_ps_vs_pro_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_pro_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_hbm = pd.DataFrame(mantel_res_its_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_hbm_ps_vs_pro.txt', sep='\\t')\n",
    "mantel_res_its_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.671698</td>\n",
       "      <td>0.654427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>26.000000</td>\n",
       "      <td>26.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.671698   0.654427\n",
       "p      0.000200   0.000400\n",
       "n     26.000000  26.000000"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. PowerSoil Pro - Low biomass samples\n",
    "## Filter table\n",
    "table_its_lbm_biom = table_its_lbm.view(Table)\n",
    "md_round1_ps_vs_pro_df_lbm = md_round1_ps_vs_pro_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_lbm_biom.ids()) & set(md_round1_ps_vs_pro_df_lbm.index))\n",
    "md_round1_ps_vs_pro_df_lbm = md_round1_ps_vs_pro_df_lbm.reindex(shared_)\n",
    "table_its_lbm_biom_ps_vs_pro = table_its_lbm_biom.filter(shared_)\n",
    "keep_ = table_its_lbm_biom_ps_vs_pro.ids('observation')[table_its_lbm_biom_ps_vs_pro.sum('observation') > 0]\n",
    "table_its_lbm_biom_ps_vs_pro.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_lbm_ps_vs_pro = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_lbm_biom_ps_vs_pro)\n",
    "md_round1_ps_vs_pro_q2_lbm = q2.Metadata(md_round1_ps_vs_pro_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_lbm = 344\n",
    "dists_res_its_lbm = all_dists_no_tree(table_its_lbm_ps_vs_pro,\n",
    "                      rare_depth_its_lbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_pro_q2_dist = md_round1_ps_vs_pro_q2_lbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_pro_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_pro_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_pro_q2_dist_sub = md_round1_ps_vs_pro_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_pro_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_lbm = pd.DataFrame(mantel_res_its_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_lbm_ps_vs_pro.txt', sep='\\t')\n",
    "mantel_res_its_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.684379</td>\n",
       "      <td>0.592883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.020996</td>\n",
       "      <td>0.062987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Jaccard      RPCA\n",
       "corr  0.684379  0.592883\n",
       "p     0.020996  0.062987\n",
       "n     6.000000  6.000000"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Norgen - High biomass samples\n",
    "## Filter table\n",
    "table_its_hbm_biom = table_its_hbm.view(Table)\n",
    "md_round1_ps_vs_norgen_df_hbm = md_round1_ps_vs_norgen_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_hbm_biom.ids()) & set(md_round1_ps_vs_norgen_df_hbm.index))\n",
    "md_round1_ps_vs_norgen_df_hbm = md_round1_ps_vs_norgen_df_hbm.reindex(shared_)\n",
    "table_its_hbm_biom_ps_vs_norgen = table_its_hbm_biom.filter(shared_)\n",
    "keep_ = table_its_hbm_biom_ps_vs_norgen.ids('observation')[table_its_hbm_biom_ps_vs_norgen.sum('observation') > 0]\n",
    "table_its_hbm_biom_ps_vs_norgen.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_hbm_ps_vs_norgen = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_hbm_biom_ps_vs_norgen)\n",
    "md_round1_ps_vs_norgen_q2_hbm = q2.Metadata(md_round1_ps_vs_norgen_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_hbm = 1491\n",
    "dists_res_its_hbm = all_dists_no_tree(table_its_hbm_ps_vs_norgen,\n",
    "                      rare_depth_its_hbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_norgen_q2_dist = md_round1_ps_vs_norgen_q2_hbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_norgen_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_norgen_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_norgen_q2_dist_sub = md_round1_ps_vs_norgen_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_norgen_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_hbm = pd.DataFrame(mantel_res_its_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_hbm_ps_vs_norgen.txt', sep='\\t')\n",
    "mantel_res_its_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Distance matrices must have at least 3 matching IDs between them (i.e., minimum 3x3 in size).",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/ipykernel_20918/472188758.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     32\u001b[0m     \u001b[0mmd_round1_ps_vs_norgen_q2_dist_sub\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmd_round1_ps_vs_norgen_q2_dist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdist_mantel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m     \u001b[0;31m# corr, p, n\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 34\u001b[0;31m     mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n\u001b[0m\u001b[1;32m     35\u001b[0m                                          \u001b[0mmd_round1_ps_vs_norgen_q2_dist_sub\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     36\u001b[0m                                          \u001b[0mgrouping\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Mycelium/ipynb/assets/util_updated.py\u001b[0m in \u001b[0;36mmantel_matched\u001b[0;34m(dist, mf_dist, grouping, ids, permutations)\u001b[0m\n\u001b[1;32m     62\u001b[0m         \u001b[0mdists\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdist_tmp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m     \u001b[0;31m# run mantel\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m     return mantel(dists[0], dists[1],\n\u001b[0m\u001b[1;32m     65\u001b[0m                   permutations=permutations)\n\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/qiime2-2021.11/lib/python3.8/site-packages/skbio/stats/distance/_mantel.py\u001b[0m in \u001b[0;36mmantel\u001b[0;34m(x, y, method, permutations, alternative, strict, lookup)\u001b[0m\n\u001b[1;32m    270\u001b[0m     \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    271\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 272\u001b[0;31m         raise ValueError(\"Distance matrices must have at least 3 matching IDs \"\n\u001b[0m\u001b[1;32m    273\u001b[0m                          \"between them (i.e., minimum 3x3 in size).\")\n\u001b[1;32m    274\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Distance matrices must have at least 3 matching IDs between them (i.e., minimum 3x3 in size)."
     ]
    }
   ],
   "source": [
    "# PowerSoil vs. Norgen - High biomass samples\n",
    "## Filter table\n",
    "table_its_lbm_biom = table_its_lbm.view(Table)\n",
    "md_round1_ps_vs_norgen_df_lbm = md_round1_ps_vs_norgen_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_lbm_biom.ids()) & set(md_round1_ps_vs_norgen_df_lbm.index))\n",
    "md_round1_ps_vs_norgen_df_lbm = md_round1_ps_vs_norgen_df_lbm.reindex(shared_)\n",
    "table_its_lbm_biom_ps_vs_norgen = table_its_lbm_biom.filter(shared_)\n",
    "keep_ = table_its_lbm_biom_ps_vs_norgen.ids('observation')[table_its_lbm_biom_ps_vs_norgen.sum('observation') > 0]\n",
    "table_its_lbm_biom_ps_vs_norgen.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_lbm_ps_vs_norgen = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_lbm_biom_ps_vs_norgen)\n",
    "md_round1_ps_vs_norgen_q2_lbm = q2.Metadata(md_round1_ps_vs_norgen_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_lbm = 344\n",
    "dists_res_its_lbm = all_dists_no_tree(table_its_lbm_ps_vs_norgen,\n",
    "                      rare_depth_its_lbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round1_ps_vs_norgen_q2_dist = md_round1_ps_vs_norgen_q2_lbm.to_dataframe().copy()\n",
    "md_round1_ps_vs_norgen_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round1_ps_vs_norgen_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round1_ps_vs_norgen_q2_dist_sub = md_round1_ps_vs_norgen_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round1_ps_vs_norgen_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_lbm = pd.DataFrame(mantel_res_its_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_lbm_ps_vs_norgen.txt', sep='\\t')\n",
    "mantel_res_its_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.704506</td>\n",
       "      <td>0.711708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>36.000000</td>\n",
       "      <td>35.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.704506   0.711708\n",
       "p      0.000200   0.000200\n",
       "n     36.000000  35.000000"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. MagMAX Microbiome - High biomass samples\n",
    "## Filter table\n",
    "table_its_hbm_biom = table_its_hbm.view(Table)\n",
    "md_round2_ps_vs_magmax_df_hbm = md_round2_ps_vs_magmax_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_hbm_biom.ids()) & set(md_round2_ps_vs_magmax_df_hbm.index))\n",
    "md_round2_ps_vs_magmax_df_hbm = md_round2_ps_vs_magmax_df_hbm.reindex(shared_)\n",
    "table_its_hbm_biom_ps_vs_magmax = table_its_hbm_biom.filter(shared_)\n",
    "keep_ = table_its_hbm_biom_ps_vs_magmax.ids('observation')[table_its_hbm_biom_ps_vs_magmax.sum('observation') > 0]\n",
    "table_its_hbm_biom_ps_vs_magmax.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_hbm_ps_vs_magmax = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_hbm_biom_ps_vs_magmax)\n",
    "md_round2_ps_vs_magmax_q2_hbm = q2.Metadata(md_round2_ps_vs_magmax_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_hbm = 1491\n",
    "dists_res_its_hbm = all_dists_no_tree(table_its_hbm_ps_vs_magmax,\n",
    "                      rare_depth_its_hbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_magmax_q2_dist = md_round2_ps_vs_magmax_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_magmax_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_magmax_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_magmax_q2_dist_sub = md_round2_ps_vs_magmax_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_magmax_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_hbm = pd.DataFrame(mantel_res_its_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_hbm_ps_vs_magmax.txt', sep='\\t')\n",
    "mantel_res_its_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.493647</td>\n",
       "      <td>0.308133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.002999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>33.000000</td>\n",
       "      <td>33.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.493647   0.308133\n",
       "p      0.000200   0.002999\n",
       "n     33.000000  33.000000"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. MagMAX Microbiome - Low biomass samples\n",
    "## Filter table\n",
    "table_its_lbm_biom = table_its_lbm.view(Table)\n",
    "md_round2_ps_vs_magmax_df_lbm = md_round2_ps_vs_magmax_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_lbm_biom.ids()) & set(md_round2_ps_vs_magmax_df_lbm.index))\n",
    "md_round2_ps_vs_magmax_df_lbm = md_round2_ps_vs_magmax_df_lbm.reindex(shared_)\n",
    "table_its_lbm_biom_ps_vs_magmax = table_its_lbm_biom.filter(shared_)\n",
    "keep_ = table_its_lbm_biom_ps_vs_magmax.ids('observation')[table_its_lbm_biom_ps_vs_magmax.sum('observation') > 0]\n",
    "table_its_lbm_biom_ps_vs_magmax.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_lbm_ps_vs_magmax = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_lbm_biom_ps_vs_magmax)\n",
    "md_round2_ps_vs_magmax_q2_lbm = q2.Metadata(md_round2_ps_vs_magmax_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_lbm = 344\n",
    "dists_res_its_lbm = all_dists_no_tree(table_its_lbm_ps_vs_magmax,\n",
    "                      rare_depth_its_lbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_magmax_q2_dist = md_round2_ps_vs_magmax_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_magmax_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_magmax_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_magmax_q2_dist_sub = md_round2_ps_vs_magmax_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_magmax_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_lbm = pd.DataFrame(mantel_res_its_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_lbm_ps_vs_magmax.txt', sep='\\t')\n",
    "mantel_res_its_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.575869</td>\n",
       "      <td>0.64412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.00020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>33.000000</td>\n",
       "      <td>32.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard      RPCA\n",
       "corr   0.575869   0.64412\n",
       "p      0.000200   0.00020\n",
       "n     33.000000  32.00000"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. NucleoMag Food - High biomass samples\n",
    "## Filter table\n",
    "table_its_hbm_biom = table_its_hbm.view(Table)\n",
    "md_round2_ps_vs_nucleomag_df_hbm = md_round2_ps_vs_nucleomag_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_hbm_biom.ids()) & set(md_round2_ps_vs_nucleomag_df_hbm.index))\n",
    "md_round2_ps_vs_nucleomag_df_hbm = md_round2_ps_vs_nucleomag_df_hbm.reindex(shared_)\n",
    "table_its_hbm_biom_ps_vs_nucleomag = table_its_hbm_biom.filter(shared_)\n",
    "keep_ = table_its_hbm_biom_ps_vs_nucleomag.ids('observation')[table_its_hbm_biom_ps_vs_nucleomag.sum('observation') > 0]\n",
    "table_its_hbm_biom_ps_vs_nucleomag.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_hbm_ps_vs_nucleomag = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_hbm_biom_ps_vs_nucleomag)\n",
    "md_round2_ps_vs_nucleomag_q2_hbm = q2.Metadata(md_round2_ps_vs_nucleomag_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_hbm = 1491\n",
    "dists_res_its_hbm = all_dists_no_tree(table_its_hbm_ps_vs_nucleomag,\n",
    "                      rare_depth_its_hbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_nucleomag_q2_dist = md_round2_ps_vs_nucleomag_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_nucleomag_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_nucleomag_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_nucleomag_q2_dist_sub = md_round2_ps_vs_nucleomag_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_nucleomag_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_hbm = pd.DataFrame(mantel_res_its_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_hbm_ps_vs_nucleomag.txt', sep='\\t')\n",
    "mantel_res_its_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.497391</td>\n",
       "      <td>0.242143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.022595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>32.000000</td>\n",
       "      <td>32.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.497391   0.242143\n",
       "p      0.000200   0.022595\n",
       "n     32.000000  32.000000"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. NucleoMag Food - Low biomass samples\n",
    "## Filter table\n",
    "table_its_lbm_biom = table_its_lbm.view(Table)\n",
    "md_round2_ps_vs_nucleomag_df_lbm = md_round2_ps_vs_nucleomag_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_lbm_biom.ids()) & set(md_round2_ps_vs_nucleomag_df_lbm.index))\n",
    "md_round2_ps_vs_nucleomag_df_lbm = md_round2_ps_vs_nucleomag_df_lbm.reindex(shared_)\n",
    "table_its_lbm_biom_ps_vs_nucleomag = table_its_lbm_biom.filter(shared_)\n",
    "keep_ = table_its_lbm_biom_ps_vs_nucleomag.ids('observation')[table_its_lbm_biom_ps_vs_nucleomag.sum('observation') > 0]\n",
    "table_its_lbm_biom_ps_vs_nucleomag.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_lbm_ps_vs_nucleomag = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_lbm_biom_ps_vs_nucleomag)\n",
    "md_round2_ps_vs_nucleomag_q2_lbm = q2.Metadata(md_round2_ps_vs_nucleomag_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_lbm = 344\n",
    "dists_res_its_lbm = all_dists_no_tree(table_its_lbm_ps_vs_nucleomag,\n",
    "                      rare_depth_its_lbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_nucleomag_q2_dist = md_round2_ps_vs_nucleomag_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_nucleomag_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_nucleomag_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_nucleomag_q2_dist_sub = md_round2_ps_vs_nucleomag_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_nucleomag_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_lbm = pd.DataFrame(mantel_res_its_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_lbm_ps_vs_nucleomag.txt', sep='\\t')\n",
    "mantel_res_its_lbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.735241</td>\n",
       "      <td>0.7349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>45.000000</td>\n",
       "      <td>45.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard     RPCA\n",
       "corr   0.735241   0.7349\n",
       "p      0.000200   0.0002\n",
       "n     45.000000  45.0000"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Zymo MagBead - High biomass samples\n",
    "## Filter table\n",
    "table_its_hbm_biom = table_its_hbm.view(Table)\n",
    "md_round2_ps_vs_zymo_df_hbm = md_round2_ps_vs_zymo_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_hbm_biom.ids()) & set(md_round2_ps_vs_zymo_df_hbm.index))\n",
    "md_round2_ps_vs_zymo_df_hbm = md_round2_ps_vs_zymo_df_hbm.reindex(shared_)\n",
    "table_its_hbm_biom_ps_vs_zymo = table_its_hbm_biom.filter(shared_)\n",
    "keep_ = table_its_hbm_biom_ps_vs_zymo.ids('observation')[table_its_hbm_biom_ps_vs_zymo.sum('observation') > 0]\n",
    "table_its_hbm_biom_ps_vs_zymo.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_hbm_ps_vs_zymo = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_hbm_biom_ps_vs_zymo)\n",
    "md_round2_ps_vs_zymo_q2_hbm = q2.Metadata(md_round2_ps_vs_zymo_df_hbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_hbm = 1491\n",
    "dists_res_its_hbm = all_dists_no_tree(table_its_hbm_ps_vs_zymo,\n",
    "                      rare_depth_its_hbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_zymo_q2_dist = md_round2_ps_vs_zymo_q2_hbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_zymo_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_zymo_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_hbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_hbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_zymo_q2_dist_sub = md_round2_ps_vs_zymo_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_hbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_zymo_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_hbm = pd.DataFrame(mantel_res_its_hbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_hbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_hbm_ps_vs_zymo.txt', sep='\\t')\n",
    "mantel_res_its_hbm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Jaccard</th>\n",
       "      <th>RPCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>corr</th>\n",
       "      <td>0.230052</td>\n",
       "      <td>0.183615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p</th>\n",
       "      <td>0.021996</td>\n",
       "      <td>0.096581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n</th>\n",
       "      <td>19.000000</td>\n",
       "      <td>19.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Jaccard       RPCA\n",
       "corr   0.230052   0.183615\n",
       "p      0.021996   0.096581\n",
       "n     19.000000  19.000000"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PowerSoil vs. Zymo MagBead - Low biomass samples\n",
    "## Filter table\n",
    "table_its_lbm_biom = table_its_lbm.view(Table)\n",
    "md_round2_ps_vs_zymo_df_lbm = md_round2_ps_vs_zymo_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_lbm_biom.ids()) & set(md_round2_ps_vs_zymo_df_lbm.index))\n",
    "md_round2_ps_vs_zymo_df_lbm = md_round2_ps_vs_zymo_df_lbm.reindex(shared_)\n",
    "table_its_lbm_biom_ps_vs_zymo = table_its_lbm_biom.filter(shared_)\n",
    "keep_ = table_its_lbm_biom_ps_vs_zymo.ids('observation')[table_its_lbm_biom_ps_vs_zymo.sum('observation') > 0]\n",
    "table_its_lbm_biom_ps_vs_zymo.filter(keep_, axis='observation')\n",
    "\n",
    "## Import filtered table and re-indexed metadata file\n",
    "table_its_lbm_ps_vs_zymo = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_lbm_biom_ps_vs_zymo)\n",
    "md_round2_ps_vs_zymo_q2_lbm = q2.Metadata(md_round2_ps_vs_zymo_df_lbm)\n",
    "\n",
    "## Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its_lbm = 344\n",
    "dists_res_its_lbm = all_dists_no_tree(table_its_lbm_ps_vs_zymo,\n",
    "                      rare_depth_its_lbm)\n",
    "\n",
    "## Make a unique ID\n",
    "md_round2_ps_vs_zymo_q2_dist = md_round2_ps_vs_zymo_q2_lbm.to_dataframe().copy()\n",
    "md_round2_ps_vs_zymo_q2_dist['unique_sample_id'] = ['.'.join(rn_.split('.')[:-2])\n",
    "                               for rn_ in md_round2_ps_vs_zymo_q2_dist.index]\n",
    "grouping = 'extraction_kit'\n",
    "ids = 'unique_sample_id'\n",
    "\n",
    "## Run Mantel test for each distance matrix\n",
    "mantel_res_its_lbm = {}\n",
    "for metric_, dist_mantel in dists_res_its_lbm.items():\n",
    "    # subset mf for dist (rare)\n",
    "    dist_mantel = dist_mantel.distance_matrix.view(DistanceMatrix)\n",
    "    md_round2_ps_vs_zymo_q2_dist_sub = md_round2_ps_vs_zymo_q2_dist.reindex(dist_mantel.ids)\n",
    "    # corr, p, n\n",
    "    mantel_res_its_lbm[metric_] = mantel_matched(dist_mantel,\n",
    "                                         md_round2_ps_vs_zymo_q2_dist_sub,\n",
    "                                         grouping,\n",
    "                                         ids)\n",
    "\n",
    "## Compile \n",
    "mantel_res_its_lbm = pd.DataFrame(mantel_res_its_lbm,\n",
    "                          ['corr', 'p', 'n'])\n",
    "mantel_res_its_lbm.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/mantel_correlations/table_mantel_its_lbm_ps_vs_zymo.txt', sep='\\t')\n",
    "mantel_res_its_lbm\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Stepwise ANOVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sample_name\n",
       "skin.nares.A.2.1.microbiome2.C                    skin.nares.A.2.1\n",
       "built.doorknob.D.2.microbiome20.C               built.doorknob.D.2\n",
       "skin.forehead.eluent.P.1.microbiome2.C    skin.forehead.eluent.P.1\n",
       "uriM.O.2.microbiome20.L                                   uriM.O.2\n",
       "oral.throat.eluent.U.microbiome20.C           oral.throat.eluent.U\n",
       "                                                    ...           \n",
       "wat.S.A.1.microbiome20.L                                 wat.S.A.1\n",
       "skin.nares.B.1.1.microbiome2.C                    skin.nares.B.1.1\n",
       "oral.saline.vtm.A.3.microbiome20.C             oral.saline.vtm.A.3\n",
       "skin.forehead.B.3.2.microbiome20.C             skin.forehead.B.3.2\n",
       "skin.nares.D.1.1.microbiome2.C                    skin.nares.D.1.1\n",
       "Name: unique_sample_id, Length: 576, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf_dist[ids]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get ordinations (row=samples, cols=axes)\n",
    "pcoa_res = {}\n",
    "pcoa_res['Jaccard'] = pcoa(dists_res['Jaccard'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['Unweighted UniFrac'] = pcoa(dists_res['Unweighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['Weighted UniFrac'] = pcoa(dists_res['Weighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res['RPCA'] = dists_res['RPCA'].biplot.view(OrdinationResults).samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /home/mestaki/Desktop/post-doc/magmax_comparison/assets/stepwise-rda.R /tmp/tmp1sqqp8k4/ord_.tsv /tmp/tmp1sqqp8k4/mf_.txt /tmp/tmp1sqqp8k4/output.effect.size.tsv\n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /home/mestaki/Desktop/post-doc/magmax_comparison/assets/stepwise-rda.R /tmp/tmpg10q06hu/ord_.tsv /tmp/tmpg10q06hu/mf_.txt /tmp/tmpg10q06hu/output.effect.size.tsv\n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /home/mestaki/Desktop/post-doc/magmax_comparison/assets/stepwise-rda.R /tmp/tmpoguo7bmz/ord_.tsv /tmp/tmpoguo7bmz/mf_.txt /tmp/tmpoguo7bmz/output.effect.size.tsv\n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /home/mestaki/Desktop/post-doc/magmax_comparison/assets/stepwise-rda.R /tmp/tmps89o6dpj/ord_.tsv /tmp/tmps89o6dpj/mf_.txt /tmp/tmps89o6dpj/output.effect.size.tsv\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>R2.adj</th>\n",
       "      <th>Df</th>\n",
       "      <th>AIC</th>\n",
       "      <th>F</th>\n",
       "      <th>Pr(&gt;F)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Jaccard</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.907956</td>\n",
       "      <td>85</td>\n",
       "      <td>-444.599389</td>\n",
       "      <td>47.768527</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ incubation</th>\n",
       "      <td>0.006910</td>\n",
       "      <td>1</td>\n",
       "      <td>-475.401064</td>\n",
       "      <td>26.811670</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Unweighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.928937</td>\n",
       "      <td>85</td>\n",
       "      <td>-549.116900</td>\n",
       "      <td>62.976966</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ incubation</th>\n",
       "      <td>0.002175</td>\n",
       "      <td>1</td>\n",
       "      <td>-560.948711</td>\n",
       "      <td>11.041108</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Weighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.915989</td>\n",
       "      <td>85</td>\n",
       "      <td>-481.495494</td>\n",
       "      <td>52.694256</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ biomass_sample</th>\n",
       "      <td>0.003247</td>\n",
       "      <td>1</td>\n",
       "      <td>-496.691615</td>\n",
       "      <td>13.784276</td>\n",
       "      <td>0.004399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ bead_beating</th>\n",
       "      <td>0.001834</td>\n",
       "      <td>1</td>\n",
       "      <td>-505.247732</td>\n",
       "      <td>8.365581</td>\n",
       "      <td>0.000400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">RPCA</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.918215</td>\n",
       "      <td>85</td>\n",
       "      <td>-492.341302</td>\n",
       "      <td>54.229845</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ bead_beating</th>\n",
       "      <td>0.002746</td>\n",
       "      <td>1</td>\n",
       "      <td>-505.409262</td>\n",
       "      <td>12.046378</td>\n",
       "      <td>0.000200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       R2.adj  Df         AIC          F  \\\n",
       "Jaccard            + sample_type_3   0.907956  85 -444.599389  47.768527   \n",
       "                   + incubation      0.006910   1 -475.401064  26.811670   \n",
       "Unweighted UniFrac + sample_type_3   0.928937  85 -549.116900  62.976966   \n",
       "                   + incubation      0.002175   1 -560.948711  11.041108   \n",
       "Weighted UniFrac   + sample_type_3   0.915989  85 -481.495494  52.694256   \n",
       "                   + biomass_sample  0.003247   1 -496.691615  13.784276   \n",
       "                   + bead_beating    0.001834   1 -505.247732   8.365581   \n",
       "RPCA               + sample_type_3   0.918215  85 -492.341302  54.229845   \n",
       "                   + bead_beating    0.002746   1 -505.409262  12.046378   \n",
       "\n",
       "                                       Pr(>F)  \n",
       "Jaccard            + sample_type_3   0.000200  \n",
       "                   + incubation      0.000200  \n",
       "Unweighted UniFrac + sample_type_3   0.000200  \n",
       "                   + incubation      0.000200  \n",
       "Weighted UniFrac   + sample_type_3   0.000200  \n",
       "                   + biomass_sample  0.004399  \n",
       "                   + bead_beating    0.000400  \n",
       "RPCA               + sample_type_3   0.000200  \n",
       "                   + bead_beating    0.000200  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "es_all = {}\n",
    "use_ = ['sample_type', 'sample_type_2','sample_type_3','biomass_sample', 'incubation','sample_technical_replicate', 'bead_beating']\n",
    "# clean up meta (only stuff to run)\n",
    "mf_ord = mf.to_dataframe().copy()\n",
    "# shit filter but works for now\n",
    "keep_ = [v_ for v_ in mf_ord.columns\n",
    "         if len(set(mf_ord[v_])) > 1 and\n",
    "         len(set(mf_ord[v_])) < mf_ord.shape[0]//2]\n",
    "mf_ord = mf_ord[keep_]\n",
    "# run stp-wise ANOVA for all ords\n",
    "for metric_, ord_ in  pcoa_res.items():\n",
    "    # get first three axes\n",
    "    ord_ = ord_[[0,1,2]]\n",
    "    ord_.columns = ['PC1','PC2','PC3']\n",
    "    # subset/match\n",
    "    mf_ord_ = mf_ord.copy()\n",
    "    shared_ids = list(set(ord_.index)\\\n",
    "                      & set(mf_ord_.index))\n",
    "    mf_ord_ = mf_ord_.loc[shared_ids,:]\n",
    "    ord_ = ord_.loc[shared_ids,:]\n",
    "    es_all[metric_] = run_stepwise_anova(ord_, mf_ord_, use_) #mf_ord_.columns)\n",
    "# concat all runs\n",
    "es_alldf = pd.concat(es_all).rename({'+ sample_type_2':'Sample Type'}, axis=0)\n",
    "es_alldf.to_csv('results/tables/effect-size_2min_20min.tsv', sep='\\t')\n",
    "es_alldf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There appears to be differences between bead_beating, biomass, and incubation with various metrics."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (qiime2-2021.11)",
   "language": "python",
   "name": "qiime2-2021.11"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
